src/equations/compressible_navier_stokes_2d.jl

```diff
@@ -437,6 +437,59 @@ end
+    norm_ = norm(normal_direction)
+    normal = normal_direction / norm_
+
+    # Compute alignment of velocity with normal direction
+    v_normal = v1 * normal[1] + v2 * normal[2]
+
+    v1_outer = v1 - 2 * v_normal * normal[1]
+    v2_outer = v2 - 2 * v_normal * normal[2]
+
+    return v1_outer, v2_outer
+end
+
+@inline function (boundary_condition::BoundaryConditionNavierStokesWall{<:SlipWall,
+                                                                        <:Adiabatic})(flux_inner,
+                                                                                      u_inner,
+                                                                                      normal::AbstractVector,
+                                                                                      x,
+                                                                                      t,
+                                                                                      operator_type::Gradient,
+                                                                                      equations::CompressibleNavierStokesDiffusion2D{GradientVariablesPrimitive})
+    v1_outer, v2_outer = velocity_symmetry_plane(normal,
+                                                 u_inner[2],
+                                                 u_inner[3])
+
+    return SVector(u_inner[1], v1_outer, v2_outer, u_inner[4])
+end
+
+@inline function (boundary_condition::BoundaryConditionNavierStokesWall{<:SlipWall,
+                                                                        <:Adiabatic})(flux_inner,
+                                                                                      u_inner,
+                                                                                      normal::AbstractVector,
+                                                                                      x,
+                                                                                      t,
+                                                                                      operator_type::Divergence,
+                                                                                      equations::CompressibleNavierStokesDiffusion2D{GradientVariablesPrimitive})
+    normal_heat_flux = boundary_condition.boundary_condition_heat_flux.boundary_value_normal_flux_function(x,
+                                                                                                           t,
+                                                                                                           equations)
+    v1_outer, v2_outer = velocity_symmetry_plane(normal,
+                                                 u_inner[2],
+                                                 u_inner[3])
+
+    _, tau_1n, tau_2n, _ = flux_inner # extract fluxes for 2nd and 3rd equations
+    normal_energy_flux = v1_outer * tau_1n + v2_outer * tau_2n +
+                         normal_heat_flux
```

Comment:
Out of curiosity, is this the standard viscous stress treatment for slip walls? I've also imposed symmetry conditions on the normal stresses (Thm 4.2 from https://doi.org/10.1016/j.jcp.2021.110723) though that implementation might make more sense for the 'GradientVariablesEntropy' specialization.

Reply:
I am actually not sure - since this is the energy flux, i just copied what is being done for the adiabatic + no-slip wall.

For the stresses, I see the argument to rotate them also to tangential only.

Reply:
Gotcha. I don't think this is something we need to do right now for this PR - if this works OK, we can use this and implement the other version in the `GradientVariablesEntropy` setting.

Reply:
Actually, I observe also for the `GradientVariablesPrimitive` slightly better error values for the herein introduced [elixir_navierstokes_freestream_symmetry.jl](https://github.com/trixi-framework/Trixi.jl/pull/2416/files#diff-5701b684c8d351cf12917370f0955cbd20d32159e9eca668b5d34a9fa84fce3a) elixir. So maybe it makes sense to use this implementation by default?